test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,272 @@
+		page,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		// Click undo in the snackbar. This reverts revert template action.
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
+
+		//Check we have dummy content.
+		const contentAfterFirstUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterFirstUndo );
+
+		// Click undo again, this time in the header. Reverts initial dummy content.
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		// Check dummy content is gone.
+		const contentAfterSecondUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterSecondUndo );
+	} );
+
+	test( 'should show the edited content after revert, clicking undo in the header toolbar, save and reload', async ( {
+		admin,
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		await templateRevertUtils.save();
+
+		await admin.visitSiteEditor();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the edited content after revert, clicking undo in the notice and reload', async ( {
+		admin,
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
+
+		await templateRevertUtils.save();
+		await admin.visitSiteEditor();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toBe( contentAfter );
+	} );
+} );
+
+class TemplateRevertUtils {
+	constructor( { editor, page } ) {
+		this.editor = editor;
+		this.page = page;
+	}
+
+	async addDummyText() {
+		await this.editor.insertBlock( { name: 'core/paragraph' } );
+		await this.page.keyboard.type( 'Test' );
```

Comment:
We can combine these together to insert a block with the text of "Test" directly.

```suggestion
		await this.editor.insertBlock({
			name: 'core/paragraph',
			attributes: { content: 'Test' },
		} );
```

Furthermore, we could argue that making it an util isn't necessary for this one-liner and we could just inline them in the tests. That's up to you though and both work fine to me.